kobo-uncaged/util.go

```diff
@@ -74,3 +75,15 @@ func readJSON(fn string, out interface{}) (emptyOrNotExist bool, err error) {
 
 	return false, json.NewDecoder(f).Decode(out)
 }
+
+// resizeKeepAspectRatioByExpanding resizes a sz to fill bounds while keeping
+// the aspect ratio. It is based on Qt::KeepAspectRatioByExpanding.
+func resizeKeepAspectRatioByExpanding(sz image.Point, bounds image.Point) image.Point {
+	if sz.X == 0 || sz.Y == 0 {
+		return sz
+	}
+	if rw := bounds.Y * sz.X / sz.Y; rw >= bounds.X {
+		return image.Pt(rw, bounds.Y)
```

Comment:
Not familiar with Go, but this technically ought to be round'ed instead of truncated (AFAICT, Qt truncates, while IM rounds).

Reply:
FWIW, that's where the mismatch between the 149x199 I got with my text script and the 149x198 Nickel generates ;).

Reply:
This is highly nitpicky, so if you prefer to honor Qt's behavior, that's perfectly fine by me ;).

Reply:
Point uses integers, so yeah, that will truncate.

Reply:
Yep, it truncates. I'm following Qt's behavior (I verified the tests with Qt's resizing).